mods/doors/init.lua

```diff
@@ -651,59 +651,99 @@ minetest.register_craft({
+	fence_closed.mesh = "doors_fencegate_closed.obj"
+	fence_closed.gate = name .. "_open"
+	fence_closed.sound = "doors_fencegate_open"
+	fence_closed.collision_box = {
 		type = "fixed",
 		fixed = {-1/2, -1/2, -1/4, 1/2, 1/2, 1/4},
-	},
-}
+	}
 
-local fence_closed = table.copy(fence)
-fence_closed.mesh = "doors_fencegate_closed.obj"
-fence_closed.gate = "doors:gate_open"
-fence_closed.sound = "doors_fencegate_open"
-fence_closed.collision_box = {
-	type = "fixed",
-	fixed = {-1/2, -1/2, -1/4, 1/2, 1/2, 1/4},
-}
+	local fence_open = table.copy(fence)
+	fence_open.mesh = "doors_fencegate_open.obj"
+	fence_open.gate = name .. "_closed"
+	fence_open.sound = "doors_fencegate_close"
+	fence_open.groups.not_in_creative_inventory = 1
+	fence_open.collision_box = {
+		type = "fixed",
+		fixed = {{-1/2, -1/2, -1/4, -3/8, 1/2, 1/4},
+			{-5/8, -3/8, -14/16, -3/8, 3/8, 0}},
+	}
 
-local fence_open = table.copy(fence)
-fence_open.mesh = "doors_fencegate_open.obj"
-fence_open.gate = "doors:gate_closed"
-fence_open.sound = "doors_fencegate_close"
-fence_open.groups.not_in_creative_inventory = 1
-fence_open.collision_box = {
-	type = "fixed",
-	fixed = {{-1/2, -1/2, -1/4, -3/8, 1/2, 1/4},
-		{-5/8, -3/8, -14/16, -3/8, 3/8, 0}},
-}
+	minetest.register_node(":" .. name .. "_closed", fence_closed)
+	minetest.register_node(":" .. name .. "_open", fence_open)
+end
+
+doors.register_fencegate("doors:gate_wood", {
+	description = "Wooden Fence Gate",
+	texture = "default_wood.png",
+	material = "default:wood",
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2}
+})
 
-minetest.register_node("doors:gate_closed", fence_closed)
-minetest.register_node("doors:gate_open", fence_open)
+doors.register_fencegate("doors:gate_acacia_wood", {
+	description = "Acacia Fence Gate",
+	texture = "default_acacia_wood.png",
+	material = "default:acacia_wood",
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2}
+})
 
-minetest.register_craft({
-	output = "doors:gate_closed",
-	recipe = {
-		{"default:stick", "group:wood", "default:stick"},
-		{"default:stick", "group:wood", "default:stick"},
-	},
+doors.register_fencegate("doors:gate_junglewood", {
+	description = "Junglewood Fence Gate",
+	texture = "default_junglewood.png",
+	material = "default:junglewood",
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2}
+})
+
+doors.register_fencegate("doors:gate_pine_wood", {
+	description = "Pine Fence Gate",
+	texture = "default_pine_wood.png",
+	material = "default:pine_wood",
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2}
+})
+
+doors.register_fencegate("doors:gate_aspen_wood", {
+	description = "Aspen Fence Gate",
+	texture = "default_aspen_wood.png",
+	material = "default:aspen_wood",
+	groups = {choppy = 2, oddly_breakable_by_hand = 2, flammable = 2}
 })
 
+minetest.register_alias("doors:gate_closed", "doors:gate_wood_closed")
+minetest.register_alias("doors:gate_open", "doors:gate_wood_open")
```

Comment:
Remove these 2 aliases please. This nodetype wasn't introduced in a previously released build, so maintaining backwards compatible code for it is somewhat unneeded.